### PR TITLE
Enhance User experience when updating Donor's information

### PR DESCRIPTION
**Pull request by alahab2**
This change improves the flow of updating a `Donor` on `donors/show`:

- A User sees `donors/edit` form on `donors/show` after he/she created a new `Donation`;
- A User can add :title, :postal_code and :comments for a `Donor`.

**Note:** This change updates the `schema`. Please run `rake db:drop db:create db:migrate db:seed`.

See the screenshots below:

---

screencapture-localhost-3000-donors-5-1480327909255

screencapture-localhost-3000-donors-6-1480329234236

**Before submitting, check that:**

 - [ ] You have added (passing) tests for your code.
 - [x] You have written good* commit messages.
 - [ ] You have squashed relevant commits together.
 - [x] You have ensured that RuboCop is passing.
 - [x] Your PR relates to one subject, with a clear title and
       description using complete, grammatically correct sentences.

---

If your change contains views, ensure that:

 - [x] You have included a screenshot of the new view.

---

*If you do not know what makes a good commit message, or why it
is important, please refer to these resources: [1][1], [2][2], [3][3].

[1]: https://robots.thoughtbot.com/5-useful-tips-for-a-better-commit-message
[2]: http://chris.beams.io/posts/git-commit/
[3]: https://github.com/blog/1943-how-to-write-the-perfect-pull-request

### CODE REVIEW

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -23,9 +23,15 @@
 
  .metadata {
    color: rgba(0,0,0,0.5);
-   margin-top: -0.6rem;
    font-size: 0.9rem;
-   font-style: italic;
+   &.italic {
```
Comment (Drenmi) [inline app/assets/stylesheets/application.scss]: There's already an actual HTML element for italics, `<i>`. 😀

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -23,7 +23,7 @@ def sortable(column)
   def whodunnit(object)
     if object.versions.last&.event == "update"
       "Last updated on #{l object.versions.last.created_at.localtime} by #{User.where('id = ?', object.versions.last.whodunnit).first.email}"
-    elsif object.versions.last&.event == "create"
+    elsif object.versions.last&.event == "create" && object.versions.last.whodunnit != nil
```
Comment (Drenmi) [inline app/helpers/application_helper.rb]: There's never a need to do `== nil` or `!= nil` in Ruby. If falsiness is enough, then just:

```
!object.versions.last.whodunnit
```

If we need a "hard" check for `nil`:

```
!object.versions.last.whodunnit.nil?
```

File: app/views/donors/_show.html.slim [hunk 3]
```diff
@@ -1,17 +1,22 @@
-.row
-  .col-xs-6
-    p Name
-    p = @donor.name
+.card
+  .card-block
+    .card-text
+      .col-xs-12
```
Comment (Drenmi) [inline app/views/donors/_show.html.slim]: Bootstrap columns must go directly inside rows. 😀

File: app/views/donors/show.html.slim [hunk 4]
```diff
@@ -1,44 +1,45 @@
...
-            td = l donation.created_at.to_date, format: donation.created_at.year == Date.today.year ? :short : :default
-            td $#{donation.amount.to_i}
-            td = donation.cause&.name
-            td
-              -if donation.event.present?
-                = link_to donation.event.name, event_path(donation.event)
-            td
-              = link_to edit_donation_path(donation), remote: true do
-                i.fa.fa-pencil.fa-lg
+.row
```
Comment (Drenmi) [inline app/views/donors/show.html.slim]: The only thing that can go directly in a Bootstrap row is a column. 😀

File: app/views/donors/show.html.slim [hunk 5]
```diff
@@ -1,44 +1,45 @@
...
+  #edit_partial
+
+- if @new_donation == "true"
+    = render "edit"
+- else
+  .row
+    #show_partial
```
Comment (Drenmi) [inline app/views/donors/show.html.slim]: The only thing that can go directly in a Bootstrap row is a column. 😀

File: app/views/donors/show.html.slim [hunk 6]
```diff
@@ -1,44 +1,45 @@
...
+      = render "show"
+
+.row
+  .card.card-inverse.card-info.text-xs-center
```
Comment (Drenmi) [inline app/views/donors/show.html.slim]: The only thing that can go directly in a Bootstrap row is a column. 😀

File: app/views/donors/show.html.slim [hunk 7]
```diff
@@ -1,44 +1,45 @@
...
+    .card-block
+      h1 Total amount donated
+      h1 $#{number_with_delimiter(@donor.total_donations(params[:cause_id]), delimiter: ",")}
+
+
+.row
+  .table-responsive
```
Comment (Drenmi) [inline app/views/donors/show.html.slim]: The only thing that can go directly in a Bootstrap row is a column. 😀

File: app/views/donors/show.html.slim [hunk 8]
```diff
@@ -1,44 +1,45 @@
...
+      thead.thead-default
+        tr
+          th Donated on
+          th Amount
+          th Cause
+          th Event
+          th
+      tbody
+
+        - if @donor.donations.present?
```
Comment (Drenmi) [inline app/views/donors/show.html.slim]: This `if` statement shouldn't be needed. 😀

File: app/views/donors/show.html.slim [hunk 9]
```diff
@@ -1,44 +1,45 @@
...
+          - @donor.donations.each do |donation|
+           tr
+              td = l donation.created_at.to_date, format: donation.created_at.year == Date.today.year ? :short : :default
+              td $#{donation.amount.to_i}
+              td = donation.cause&.name
+              td
+                -if donation.event.present?
```
Comment (Drenmi) [inline app/views/donors/show.html.slim]: Missing a space between `-` and `if`. 😉

**Comment by alahab2** [comment]
Addressed the comments 😃

**Comment by Drenmi** [comment]
🌻 🌻 🌻

**Pull request merged**